.travis.yml

```diff
@@ -74,25 +79,43 @@ before_install:
   # https://github.com/coala/coala/issues/3183
   # https://github.com/coala/coala/issues/3184
   - pip install -U py==1.4.29 pip
+  - cp requirements.txt requirements.orig
   - cat test-requirements.txt docs-requirements.txt >> requirements.txt
   - sed -i '/^-r/d' requirements.txt
   - echo 'language_check==0.8.*' >> requirements.txt
+  # On unsupported versions, coala will be installed in the `script` block
+  - >
+    if [[ $TRAVIS_PYTHON_VERSION_MINOR < 4 || $TRAVIS_PYTHON_VERSION_MAJOR == 2 ]]; then
+      sed -e '/^coala/d' requirements.txt > requirements.new
+      mv requirements.new requirements.txt
+    fi
 
 before_script:
+  - mv requirements.orig requirements.txt
```

Comment:
setup.py reads `requirements.txt` , and needs the original, without all of the fiddles done above.